vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "iCR" ~ 8,
+    arg == "iPR" ~ 7,
+    arg == "iSD" ~ 6,
+    arg == "NON-iCR/NON-iUPD" ~ 5,
+    arg == "iCPD" ~ 4,
+    arg == "iUPD" ~ 3,
+    arg == "NE" ~ 2,
+    arg == "MISSING" ~ 1,
+    TRUE ~ NA_real_
+  )
+}
+
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp_new(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, AVALC, AVAL)
+)
+```
+
+### Flag Worst Assessment at Each Date (`ANL01FL`) {#anl01fl}
+
+When deriving `ANL01FL` this is an opportunity to exclude any records
+that should not contribute to any downstream parameter derivations. In
+the below example this includes only selecting valid assessments and
+those occurring on or after randomization date. If there is more than
+one assessment at a date, the worst one is flagged.
+
+```{r}
+adrs <- adrs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "last"
+    ),
+    filter = !is.na(AVAL) & ADT >= RANDDT
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL)
+)
+```
+
+Here is an alternative example where those records occurring after new
+anti-cancer therapy are additionally excluded (where `NACTDT` would be
+pre-derived as first date of new anti-cancer therapy. See `{admiralonco}`
+[Creating and Using New Anti-Cancer Start Date](nactdt.html) for deriving this
+variable).
+
+```{r, eval=FALSE}
+adrs <- adrs %>%
+  mutate(
+    ANL01FL = case_when(
+      !is.na(AVAL) & ADT >= RANDDT & ADT < NACTDT ~ "Y",
+      TRUE ~ NA_character_
+    )
+  )
+```
+
+### Flag Assessments up to First PD (`ANL02FL`) {#anl02fl}
+
+To restrict response data up to and including first reported progressive disease
+`ANL02FL` flag could be created by using `{admiral}` function
+`admiral::derive_var_relative_flag()`.
+
+```{r}
+adrs <- adrs %>%
+  derive_var_relative_flag(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(ADT, RSSEQ),
+    new_var = ANL02FL,
+    condition = AVALC == "iCPD",
+    mode = "first",
+    selection = "before",
+    inclusive = TRUE
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, AVALC, ADT, ANL01FL, ANL02FL)
+)
+```
+
+### Select Source Assessments for Parameter derivations
+
+For most parameter derivations the post-baseline overall response assessments up
+to and including first PD are considered.
+```{r}
+ovr <- filter(adrs, PARAMCD == "OVR" & ANL01FL == "Y" & ANL02FL == "Y")
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  ovr,
+  display_vars = exprs(USUBJID, AVISIT, AVALC, ADT, RANDDT)
+)
+```
+
+## Define Events
```

Comment:
Confirmation events are defined at this place in adrs.html, but many events were already predefined.

We can either define all events here, i.e. in addition please add above:
irsp_y
icb_y
ibor_icr
ibor_ipr
ibor_isd
ibor_non_icripd
ibor_icpd
ibor_iupd
ibor_ne

The second solution is to define events directly before creating parameters. Then the 4 definitions (please add icrsp_y_cr, icrsp_y_pr definitions as well) would have to be moved to the section "Derive Response Parameters requiring Confirmation".

Reply:
will keep before param